### ISSUE TITLE
If Nodes / Spans GeoJSOn uploaded and conversion to JSON crashes, show nice messages to user

### ISSUE THREAD

**Issue by ghost** [Open-Telecoms-Data/cove-ofds#84]
Think this is what happened with https://github.com/Open-Telecoms-Data/lib-cove-ofds/issues/69

We should fix in library but also in Cove add some crash protection

**Comment by duncandewhurst** [comment]
I think the ultimate solution to this might be:

* https://github.com/Open-Telecoms-Data/open-fibre-data-standard/issues/253

**Comment by ghost** [comment]
The only way I see a GeoJSON Schema removing the need for this work is if we validate GeoJSON before conversion and refuse to process if not perfect. That doesn't seem a great option. There could still be errors that we can't protect against in JSONSchema and it seems generally it would be better to try and be forgiving where possible. (A GeoJSON schema may be neat for other reasons but I don't think it helps here)

**Comment by ghost** [comment]
![Screenshot from 2023-02-17 11-44-02](https://user-images.githubusercontent.com/37832009/219645745-3d0a7a9e-5a20-4c6d-b57c-c1f38bdf833b.png)

This is the current screen that comes up - I think this might already be done actually?

**Comment by duncandewhurst** [comment]
That's better than showing the Python error but it won't help users troubleshoot problems with the data that they submitted like the one in https://github.com/Open-Telecoms-Data/lib-cove-ofds/issues/69#issuecomment-1422389794

**Comment by ghost** [comment]
Yes - the first part of that work is making the library more robust so it defensively checks data and doesn't crash on bad data. We're doing that in issues like the one you linked. (Fix already in lib but not live yet).

Then it's seeing how the conversion can raise nice messages to the users. The GeoJSON -> JSON library already reports on problems like inconsistent_network_ids, so it's putting more checks in there and raising those messages to the users. Maybe a schema for GeoJSON can help with that? (tho that doesn't remove the need for other work)

Are you thinking there is other things we could do here? Maybe a chat to go over ideas would be best.
